### DISCUSSION TITLE
How to change colors and number of bins in histogram(grains)?

### DISCUSSION THREAD

**Discussion by AdamLeaghj** [mtex-toolbox/mtex#1704] [Ask Anything] [answered]
Hi everyone,

Just like the title, I used `histogram(grains('Indexed'))` to plot the grain size distribution in a material consisting of 2 phases. How can I change the color of each phase and change the number of bins in this chart? Thank you!

![image](https://user-images.githubusercontent.com/118141541/210858962-17c3502c-9643-4e00-803d-70229826290f.png)

**Comment by kilir** [comment, marked as answer]
Hi,
have a look at `help grain2d/histogram`.

```
number_of_bins = 13
grains = ebsd.calcGrains
h = histogram(grains('p','d'),number_of_bins)
h(1).FaceColor = [0.8 0.5 0.2]
h(2).FaceColor = [0.2 0.7 0.3]
```
Hope that helps.
Cheers,
Rüdiger

**Reply by AdamLeaghj** [reply]
Hi Rudiger,

Thanks for your help. It works perfectly! Just a follow-up question: it seems like the fraction of the FCC phase is overlapped with the BCC (see image below). How can I change to a stacked type? Thanks.

![image](https://user-images.githubusercontent.com/118141541/210884038-6f987def-ac73-411c-b724-b15b5522370e.png)

**Reply by kilir** [reply]
Have  a look at `bar()`:
```
h = histogram(grains)
bar(h(1).BinEdges(1:end-1) + h(1).BinWidth/2, [h(1).Values; h(2).Values])
```
Cheers,
Rüdiger

**Reply by AdamLeaghj** [reply]
Thanks Rudiger!!!
